### PR TITLE
Rename Openswitch to Libreswitch

### PR DESCRIPTION
**Pull request by skg-net**
##### SUMMARY
The old openswitch(ops) (HP) NOS has been migrated to Libreswitch. There is now a new openswitch (opx) NOS supported by Dell and other vendors.
Hence moving the old openswitch modules to libreswitch folders.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 module/network/openswitch
 module/network/libreswitch

##### ANSIBLE VERSION
ansible  --version
ansible 2.4.0 (libreswitch a86782e1ba) last updated 2017/08/09 09:22:17 (GMT -700)
  config file = None
  configured module search path = [u'/home/sganesan/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/sganesan/github/ansible/lib/ansible
  executable location = /home/sganesan/github/ansible/bin/ansible
  python version = 2.7.11 |Continuum Analytics, Inc.| (default, Dec  6 2015, 18:08:32) [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)]
##### ADDITIONAL INFORMATION
- Currently the modules names remain ops
      * the new openswitch have a different short form opx
      * all the existing tests/playbooks needs to change.
      * libreswitch community haven't got back on the naming convention.

- I haven't tested this changes, since libreswitch image is not available currently.

### CODE REVIEW

**Comment by skg-net** [comment]
@gundalow  @privateip Please have a look at it.

**Comment by Qalthos** [comment]
@skg-net are you planning on maintaining this module and addressing issues with it in the future? I am given to understand that even without the name change, these modules were not working properly.

**Comment by skg-net** [comment]
@Qalthos No, I am not planning to own/maintain these modules. My understanding is if you try with new openswitch modules, it wont work, since its a completely different NOS. But if you try to run with the old openswitch (now libreswitch) it should work. 

Since we are going to upload the new openswitch modules, I am just trying to cleanup the namespace.

File: docs/docsite/rst/dev_guide/developing_module_utilities.rst [hunk 1]
```diff
@@ -38,7 +38,7 @@ The following is a list of module_utils files and a general description. The mod
 - network.py - Functions for running commands on networking devices
 - nxos.py - Contains definitions and helper functions specific to Cisco NXOS networking devices
 - openstack.py - Utilities for modules that work with Openstack instances.
-- openswitch.py - Definitions and helper functions for modules that manage OpenSwitch devices
+- libreswitch.py - Definitions and helper functions for modules that manage libreswitch devices
```
Comment (gundalow) [inline docs/docsite/rst/dev_guide/developing_module_utilities.rst:41]: This list should be in alphabetical order.

File: test/sanity/pep8/legacy-files.txt [hunk 2]
```diff
@@ -349,9 +349,9 @@ lib/ansible/modules/network/nxos/nxos_vtp_domain.py
 lib/ansible/modules/network/nxos/nxos_vtp_password.py
 lib/ansible/modules/network/nxos/nxos_vtp_version.py
 lib/ansible/modules/net_tools/omapi_host.py
-lib/ansible/modules/network/openswitch/ops_command.py
-lib/ansible/modules/network/openswitch/ops_config.py
-lib/ansible/modules/network/openswitch/ops_facts.py
+lib/ansible/modules/network/libreswitch/ops_command.py
```
Comment (gundalow) [inline test/sanity/pep8/legacy-files.txt:352]: Alphabetical order

**Review by gundalow — COMMENTED** [review]
These modules will still be called `ops_facts`

What will the new modules be called?

**Comment by skg-net** [comment]
@gundalow The new modules will be called opx_base_xxxx.

File: lib/ansible/modules/network/libreswitch/ops_command.py [hunk 3]
```diff
@@ -126,7 +126,7 @@
 """
 import traceback
 
-import ansible.module_utils.openswitch
+import ansible.module_utils.libreswitch
```
Comment (bcoca) [inline lib/ansible/modules/network/libreswitch/ops_command.py:129]: too keep compatibility for those using the old name of the lib, this should be a try/except Import with fallback to the old name.

**Review by bcoca — CHANGES_REQUESTED** [review]

**Comment by bcoca** [comment]
any plans on renaming the modules themselves? ops_ vs lbs_ ?

**Comment by Qalthos** [comment]
Since you have neither tested these modules nor do you have any interest in maintaining these modules, we are not going to merge this PR. Good news for you, though is that we have decided instead to just get rid of the modules entirely.

This was decided as the modules do not work at all presently. That being said, if anyone in the future wishes to use this as a base for new LibreSwitch modules, the files are still in git history available for retrofitting.

**Pull request closed without merging**